pkg/cmd/controller/controller_build.go

```diff
@@ -789,30 +789,38 @@ func updateForStage(si *tekton.StageInfo, a *v1.PipelineActivity) {
 	containersTerminated := false
 
 	if si.Pod != nil {
+		var stageSteps []v1.CoreActivityStep
 		pod := si.Pod
 		_, containerStatuses, _ := kube.GetContainersWithStatusAndIsInit(pod)
 		containersTerminated = len(containerStatuses) > 0
-		for _, c := range containerStatuses {
-			name := strings.Replace(strings.TrimPrefix(c.Name, "build-step-"), "-", " ", -1)
-			name = strings.Replace(strings.TrimPrefix(c.Name, "step-"), "-", " ", -1)
-			title := strings.Title(name)
+		for i, c := range containerStatuses {
+			title := getStepTitle(c.Name)
 			step, _ := kube.GetOrCreateStepInStage(stage, title)
 			running := c.State.Running
 			terminated := c.State.Terminated
 
 			var startedAt metav1.Time
 			var finishedAt metav1.Time
+			if startedAt.IsZero() {
```

Comment:
Could this not be extracted in a function which describes what's happening here? Not only do we end up with less nested code, but hopefully one also understands what's happening here.